### ISSUE TITLE
globalStrictPower and internal RF buffer

### ISSUE THREAD

**Issue by Thyriel** [Ipsis/Woot#142]
As i've noticed that the woot factory has a pretty big internal RF storage compared to the RF it's actually using (tier 1 ~10 million RF internal storage vs. 160 RF / tick usage - in other words the internal battery would last over 50 minutes) would it be possible to deactivate it's internal RF storage when using the globalStrictPower=true config ?
Doesn't make any sense storing RF when it's anyway only working when there's currently enough RF/Tick.

The problem i somehow have with that huge internal RF storage is, that in early game it's totally messing up the whole RF network for hours until it's full.

**Comment by Ipsis** [comment]
Never occured to me that the internal RF buffer would be an issue at early game.

I'll give your idea a try and see if everything still works.
You do appreciate that with strict mode if you don't have enough RF/tick it still consumes the RF but doesn't count towards the spawn?

**Comment by Thyriel** [comment]
The issue is that with an internal buffer of 10 million RF and my early game production capping at around 400 RF/tick (=8k/sec) the generators are running at full power for an hour until the internal buffer is full. All other machines connected to it (that do not use much power at all at this early stage) have a hard time getting enough power until the internal buffer is full.
And when it is full, i could even cut the power supply and the woot factory will run for 50 minutes until the internal storage gets empty, which seems (compared to other machines that last for seconds to minutes until they get empty on their own) quite a lot overhead to me.
So i thought would be great to be able to bypass that problem somehow. The globalStrictPower setting sounded like it could be usefull for that (at least i would not know why it still needs an internal power buffer with that setting). And it sounds logical that it would still consume power and just waste it if there is not enough RF/tick to spawn something.

Or if that is too complicated to implement, would it be possible to provide settings to configure the internal RF storage of each tier in the config file ?

**Comment by Ipsis** [comment]
I need to think carefully about how to handle this.
The block that stores the energy doesn't have a tier, so I risk deleting large amounts of RF from people's factories if I allow per-tier energy storage. So that is probably a no-go, but maybe a possibility for a change in 1.11.

The way, that I think, the reception of power works, is that the cables push power into the factory, rather than me pull it, therefore they need some sort of storage. So removing energy storage would probably not work. Because of that I also need to handle that wide variety of RF/tick that is needed.

As a guess, I'm probably going to lower the stored RF down a lot so that the internal buffer is a lot smaller, probably around 500,000, with an option to specify it in the config.

I just need to find a way such that current factories dont lose their stored RF as people might get upset when they upgrade to the latest version and lost 9,500,000RF!

So in summary:
* internal RF stored will be getting reduced ~500000RF
* that value will be configurable

**Comment by Thyriel** [comment]
Thanks. Imho if you want to prevent people from getting upset because of loosing 9,5M RF i think it would be fine too to just let the default RF storage where it is now and make it configurable so people like me can reduce it to what they want :)

**Comment by Ipsis** [comment]
Added the global config option to allow specifying of max energy storage.
Defaults to 10,000,000RF as before.

    # Maximum energy storage of a factory
    I:globalMaxPower=10000000

**Comment by Ipsis** [comment]
All the code is now added and it will be available in 1.10.2-1.3.0

**Comment by Thyriel** [comment]
Thanks :)
